apps/factory/src/components/create_template.tsx

```diff
@@ -74,36 +89,126 @@ export function CreateTemplate({
     try {
       onCreate?.(data);
+      console.log(data);
+      handleClose();
     } catch (err) {
       console.log(err);
     }
   };
 
+  api.prompt.getTemplate.useQuery(
+    {
+      id: `${ptId}`,
+    },
+    {
+      onSuccess(items) {
+        setDataToUpdate(items!);
+        setDefaultModelType(items?.modelType);
+      },
+    },
+  );
+
+  const fetchTemplateData = () => {
+    setIsOpen(true);
+    reset({
+      name: datatoUpdate.name,
+      description: datatoUpdate.description,
+      promptPackageId: datatoUpdate.promptPackageId,
+      modelType: datatoUpdate.modelType,
+    });
+  };
+
+  // to update
+  const mutation = api.prompt.updateTemplate.useMutation();
+
+  const updateTemplate = (data: CreateTemplateInput) => {
+    const input = {
+      id: ptId,
+      name: data.name,
+      description: data.description,
+    };
+    mutation.mutate(input, {
+      onSuccess() {
+        handleClose();
+        toast.success("Template Updated Successfully");
+        const updatedInput = {
+          name: datatoUpdate.name,
+          description: data.description,
+          promptPackageId: datatoUpdate.promptPackageId,
+          modelType: datatoUpdate.modelType,
+        };
+        setDataToUpdate(updatedInput);
+      },
+      onError(error) {
+        console.log(error);
+      },
+    });
+  };
+
   return (
     <Box component="span" sx={{}}>
       <Grid component="span">
-        <Tooltip title={"Create Template"} placement="top-start">
-          <IconButton
-            size="small"
-            aria-label="add template"
-            onClick={() => setIsOpen(true)}
-            color="primary"
-          >
-            <AddCircleIcon />
-          </IconButton>
-        </Tooltip>
+        <Grid container>
+          <Grid item xs={6} md={6} lg={6}>
+            <Tooltip title={"Create Template"} placement="top-start">
+              <IconButton
+                size="small"
+                aria-label="add template"
+                onClick={() => {
+                  setIsOpen(true);
+                  setDefaultModelType("TEXT2TEXT");
+                }}
+                color="primary"
+              >
+                <AddCircleIcon />
+              </IconButton>
+            </Tooltip>
+          </Grid>
+          {ptId ? (
+            <>
+              <Grid item xs={6} md={6} lg={6}>
+                <Tooltip title={"Edit Template"} placement="top-start">
+                  <IconButton
+                    size="small"
+                    aria-label="edit template"
+                    onClick={() => {
+                      setOpenEditTemplate(!openEditTemplate);
+                      fetchTemplateData();
+                    }}
+                    color="primary"
+                  >
+                    <EditIcon />
+                  </IconButton>
+                </Tooltip>
+              </Grid>
+            </>
+          ) : (
+            <></>
+          )}
+        </Grid>
```

Comment:
since most of the code and UI is same, please use use  `!ptId ?  <AddCircleIcon /> : <EditIcon /> ` to take care of ui.

This component should render only one button, not both, We should be using component twice if bot edit and edit are needed.

Reply:
- the addIcon will always be there as we want to add new template.
- the Edit Icon will only be shown when there is any template present so that's why I have added a check that if ptId exists then only show editIcon to edit template

Reply:
- let me try you ways